### ISSUE TITLE
[HOLD for payment 2025-02-24][$250] Expense - On selecting a category/ tags by multiple tapping, user directed to conversation page

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53344]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** V9. 0.69-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Go to workspace settings
3. Enable tags
4. Create few tags
5. Tap workspace chat
6. Create an expense with category and tag
7. Open the expense
8. Open category and select a category by multiple tapping
9. Open tag and select a tag by multiple tapping

## Expected Result:
On selecting a category/ tags by multiple tapping, user must not be directed to conversation/merchant page instead of expense details page.

## Actual Result:
On selecting a category/ tags by multiple tapping, user directed to conversation/merchant page instead of expense details page.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/fd4c73c4-0736-4313-a477-b15f23c9660c

  </details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864001016141250370</li>
        <li>Upwork Job ID: 1864001016141250370</li>
        <li>Last Price Increase: 2024-12-17</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 105632281</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alitoshmatov</details>

**Comment by puneetlath** [comment]
Bumped @alitoshmatov in Slack.

**Comment by alitoshmatov** [comment]
@Krishna2323 Can you recheck your solution, I am still able to reproduce the issue

**Comment by alitoshmatov** [comment]
Oh, my bad, I was changing tag component and testing category. Now it is working.

**Comment by alitoshmatov** [comment]
Thank you @dominictb for proposal

> We shouldn't lean on isFocused since we still can call onSelectRow multiple times when the page is focused.

Can you elaborate, are you suggesting that we can still reproduce the issue with @Krishna2323 proposal.

In your solution it seems that after first selection you just disable selecting row, can it cause some issue where we might allow another selection after first one?

**Comment by puneetlath** [comment]
@alitoshmatov can you summarize where we're at with this issue?

**Comment by alitoshmatov** [comment]
I think we can go with @Krishna2323 's [proposal](https://github.com/Expensify/App/issues/53344#issuecomment-2509396040). I think `isFocused` is a reliable state here.

C+ reviewed 🎀 👀 🎀

**Comment by alitoshmatov** [comment]
@puneetlath I reviewed proposal and selected [here](https://github.com/Expensify/App/issues/53344#issuecomment-2555819727). Please review my selection when you have time

**Comment by puneetlath** [comment]
Sorry for the delay. I'm good with going with @Krishna2323's proposal.

**Comment by puneetlath** [comment]
This was reverted, so not ready for payment.

**Comment by puneetlath** [comment]
Ok, so then I think we just need to pay out for the work that was already done. I believe we had one PR merged, but it had a regression so we reverted. Then another one was started, is that right? 

@lschurr can you handle the required payments please?

**Comment by puneetlath** [comment]
Oh wait, so what's the situation then? Did our PRs fix it? Or something else did?

**Comment by puneetlath** [comment]
Ok, no worries. That makes sense. Let's revert and then we can pay this out.

**Comment by lschurr** [comment]
Thanks @puneetlath - Just checking, there was a regression so payment should be $125, correct?

**Comment by puneetlath** [comment]
For the C+ yes, I believe that is correct.

**Comment by lschurr** [comment]
Payment summary:

- Contributor: $125 @Krishna2323 ([paid in Upwork](https://www.upwork.com/nx/wm/workroom/39440582/overview))
- Reviewer: $125 @alitoshmatov (please request in NewDot)

**Comment by JmillsExpensify** [comment]
$125 approved for @alitoshmatov
